### PR TITLE
Fix image volume storageclass error

### PR DESCRIPTION
**Pull request by futuretea**
**Related issues**
https://github.com/harvester/harvester/issues/3248

*Test plan**

- Setup test env by https://github.com/harvester/terraform-provider-harvester/wiki/PR-Test-Env-Setup for `Reviwer`
- Setup test env by https://github.com/harvester/terraform-provider-harvester/wiki/Use-the-master-head-docker-image-for-testing for `Tester`
- Create a test config file
```bash
vim main.tf
```

```hcl
resource "harvester_image" "ubuntu20" {
  name      = "ubuntu20"
  namespace = "harvester-public"

  display_name = "ubuntu-20.04-server-cloudimg-amd64.img"
  source_type  = "download"
  url          = "http://cloud-images.ubuntu.com/releases/focal/release/ubuntu-20.04-server-cloudimg-amd64.img"
}

resource "harvester_virtualmachine" "ubuntu20" {
  name                 = "ubuntu20"
  namespace            = "default"
  restart_after_update = true

  description = "test ubuntu20 raw image"
  tags = {
    ssh-user = "ubuntu"
  }

  cpu    = 2
  memory = "2Gi"

  efi         = true
  secure_boot = true

  run_strategy = "RerunOnFailure"
  hostname     = "ubuntu20"
  machine_type = "q35"

  network_interface {
    name           = "nic-1"
    wait_for_lease = true
  }

  disk {
    name       = "rootdisk"
    type       = "disk"
    size       = "10Gi"
    bus        = "virtio"
    boot_order = 1

    image       = harvester_image.ubuntu20.id
    auto_delete = true
  }

  disk {
    name        = "emptydisk"
    type        = "disk"
    size        = "20Gi"
    bus         = "virtio"
    auto_delete = true
  }

  cloudinit {
    user_data    = <<-EOF
      #cloud-config
      password: 123456
      chpasswd:
        expire: false
      ssh_pwauth: true
      package_update: true
      packages:
        - qemu-guest-agent
      runcmd:
        - - systemctl
          - enable
          - '--now'
          - qemu-guest-agent
      EOF
    network_data = ""
  }

}
```
- apply 
```bash
terraform apply -auto-approve
```
- change vm description and apply again
```bash
vim main.tf
terraform apply -auto-approve
```

### CODE REVIEW

File: internal/provider/virtualmachine/resource_virtualmachine_constructor.go [hunk 1]
```diff
@@ -199,16 +198,12 @@ func (c *Constructor) Setup() util.Processors {
...
-							pvcOption.ImageID = helper.BuildNamespacedName(imageNamespace, imageName)
-							storageClassName = builder.BuildImageStorageClassName("", imageName)
-						} else {
-							return fmt.Errorf("the %s of an image can only be defined during image creation", constants.FieldVolumeStorageClassName)
+						imageNamespace, imageName, err := helper.NamespacedNamePartsByDefault(imageNamespacedName, c.Builder.VirtualMachine.Namespace)
+						if err != nil {
+							return err
 						}
+						pvcOption.ImageID = helper.BuildNamespacedName(imageNamespace, imageName)
+						storageClassName = builder.BuildImageStorageClassName("", imageName)
```
Comment (FrankYang0529) [inline internal/provider/virtualmachine/resource_virtualmachine_constructor.go:206]: Do we want to prevent users from updating `storageClassName`?

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thanks.

**Review by chrisho — APPROVED** [review]
Verified and LGTM, thanks

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**